"make distcheck" broken on master

Tonight's nightly tarball build failed on master, due to a problem with libevent:

```
Starting build for master
Found new revision 09ad725
Exceuting run-with-autotools.sh failed:
    am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/dl/libltdl'
make[3]: Leaving directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/dl/libltdl'
(cd mca/event/external && make  top_distdir=../../../../openmpi-master-201808100242-09ad725 distdir=../../../../openmpi-master-201808100242-09ad725/opal/mca/event/external \
    am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/event/external'
make[3]: Leaving directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/event/external'
(cd mca/event/libevent2022 && make  top_distdir=../../../../openmpi-master-201808100242-09ad725 distdir=../../../../openmpi-master-201808100242-09ad725/opal/mca/event/libevent2022 \
    am__remove_distdir=: am__skip_length_check=: am__skip_mode_fix=: distdir)
make[3]: Entering directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/event/libevent2022'
make[3]: *** No rule to make target 'distdir'.  Stop.
make[3]: Leaving directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal/mca/event/libevent2022'
Makefile:2476: recipe for target 'distdir' failed
make[2]: *** [distdir] Error 1
make[2]: Leaving directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git/opal'
Makefile:2001: recipe for target 'distdir' failed
make[1]: *** [distdir] Error 1
make[1]: Leaving directory '/mnt/data/nightly-tarball/scratch/openmpi/master-201808100242/ompi.git'
Makefile:2102: recipe for target 'dist' failed
make: *** [dist] Error 2
FAILURE: <class 'subprocess.CalledProcessError'>: Command '['/home/mpiteam/ompi-scripts/nightly-tarball/run-with-autotools.sh', 'autotools/ompi-master', 'make', 'distcheck']' returned non-zero exit status 2
```

Build artifacts (including config.log and friends) are available at http://download.open-mpi.org/nightly/open-mpi/failed-builds/openmpi-master-201808100242-failed.tar.gz

It looks like the right libevent libraries are installed on the system, so with the recent change, the external libevent is being built and the libevent2022 component is not.  The build error looks to be in the libevent2022 component.

```
+++ Configuring MCA framework event
checking for no configure components in framework event... 
checking for m4 configure components in framework event... external, libevent2022

--- MCA component event:external (m4 configuration macro, priority 90)
checking for MCA component event:external compile mode... static
checking --with-libevent-libdir value... simple ok (unspecified value)
checking for external libevent in... (default search paths)
checking event.h usability... yes
checking event.h presence... yes
checking for event.h... yes
looking for library without search path
checking for library containing event_config_new... -levent
checking if libevent requires libnl v1 or v3... 
checking for evthread_set_lock_callbacks in -levent... yes
checking for evthread_use_pthreads in -levent_pthreads... yes
checking if external libevent version is 2.0.21 or greater... yes
checking if MCA component event:external can compile... yes

--- MCA component event:libevent2022 (m4 configuration macro, priority 80)
checking for MCA component event:libevent2022 compile mode... static
checking if event external component succeeded... yes
configure: event:external succeeded, so this component will be skipped
checking if MCA component event:libevent2022 can compile... no
 
checking if have working event ops for the event framework... yes
checking for winning event component header file... external/external.h
```

--- a/opal/mca/event/libevent2022/configure.m4
+++ b/opal/mca/event/libevent2022/configure.m4
@@ -14,9 +14,9 @@
 #
 AC_DEFUN([MCA_opal_event_libevent2022_PRIORITY], [80])
 
-#
-# Force this component to compile in static-only mode
-#
+dnl
+dnl Force this component to compile in static-only mode
+dnl
 AC_DEFUN([MCA_opal_event_libevent2022_COMPILE_MODE], [
     AC_MSG_CHECKING([for MCA component $2:$3 compile mode])
     $4="static"
@@ -82,9 +82,9 @@ EOF
           ])
 ])
 
-# MCA_event_libevent2022_CONFIG([action-if-can-compile],
-#                              [action-if-cant-compile])
-# ------------------------------------------------
+dnl MCA_event_libevent2022_CONFIG([action-if-can-compile],
+dnl                              [action-if-cant-compile])
+dnl ------------------------------------------------
 AC_DEFUN([MCA_opal_event_libevent2022_CONFIG],[
     # We know that the external event component will be configured
     # before this one because of its priority.  This component is only

--- a/opal/mca/event/libevent2022/configure.m4
+++ b/opal/mca/event/libevent2022/configure.m4
@@ -35,8 +35,8 @@ AC_DEFUN([MCA_opal_event_libevent2022_POST_CONFIG], [
            # copied from libevent's Makefile.am.
 
            AC_CONFIG_COMMANDS([opal/mca/event/libevent2022/libevent/include/event2/event-config.h],
-                              [libevent_basedir="opal/mca/event/libevent2022"
-                               libevent_file="$libevent_basedir/libevent/include/event2/event-config.h"
+                              [opal_event_libevent2022_basedir="opal/mca/event/libevent2022"
+                               libevent_file="$opal_event_libevent2022_basedir/libevent/include/event2/event-config.h"
                                rm -f "$libevent_file.new"
                                cat > "$libevent_file.new" <<EOF
 /* event2/event-config.h
@@ -56,7 +56,7 @@ EOF
 
                                sed -e 's/#define /#define _EVENT_/' \
                                    -e 's/#undef /#undef _EVENT_/' \
-                                   -e 's/#ifndef /#ifndef _EVENT_/' < "$libevent_basedir/libevent/config.h" >> "$libevent_file.new"
+                                   -e 's/#ifndef /#ifndef _EVENT_/' < "$opal_event_libevent2022_basedir/libevent/config.h" >> "$libevent_file.new"
                                echo "#endif" >> "$libevent_file.new"
 
                                # Only make a new .h libevent_file if the
@@ -75,7 +75,7 @@ EOF
 
            # Add some stuff to CPPFLAGS so that the rest of the source
            # tree can be built
-           libevent_file=$libevent_basedir/libevent
+           libevent_file=$opal_event_libevent2022_basedir/libevent
            CPPFLAGS="-I$OPAL_TOP_SRCDIR/$libevent_file -I$OPAL_TOP_SRCDIR/$libevent_file/include $CPPFLAGS"
            AS_IF([test "$OPAL_TOP_BUILDDIR" != "$OPAL_TOP_SRCDIR"],
                  [CPPFLAGS="-I$OPAL_TOP_BUILDDIR/$libevent_file/include $CPPFLAGS"])
@@ -87,6 +87,9 @@ dnl MCA_event_libevent2022_CONFIG([action-if-can-compile],
 dnl                              [action-if-cant-compile])
 dnl ------------------------------------------------
 AC_DEFUN([MCA_opal_event_libevent2022_CONFIG],[
+    AC_CONFIG_FILES([opal/mca/event/libevent2022/Makefile])
+    opal_event_libevent2022_basedir="opal/mca/event/libevent2022"
+
     # We know that the external event component will be configured
     # before this one because of its priority.  This component is only
     # needed if the external component was not successful in selecting
@@ -94,18 +97,37 @@ AC_DEFUN([MCA_opal_event_libevent2022_CONFIG],[
     AC_MSG_CHECKING([if event external component succeeded])
     AS_IF([test "$opal_event_external_support" = "yes"],
           [AC_MSG_RESULT([yes])
-           AC_MSG_NOTICE([event:external succeeded, so this component will be skipped])
-           $2],
+           AC_MSG_NOTICE([event:external succeeded, so this component will be configured, but then will be skipped])
+           MCA_opal_event_libevent2022_FAKE_CONFIG($2)],
           [AC_MSG_RESULT([no])
            AC_MSG_NOTICE([event:external failed, so this component will be used])
-           MCA_opal_event_libevent2022_BACKEND_CONFIG($1, $2)])
+           MCA_opal_event_libevent2022_REAL_CONFIG($1, $2)])
 ])
 
-AC_DEFUN([MCA_opal_event_libevent2022_BACKEND_CONFIG],[
-    OPAL_VAR_SCOPE_PUSH([CFLAGS_save CPPFLAGS_save libevent_file event_args libevent_happy])
+dnl
+dnl This macro is invoked when event:external is going to be used (and
+dnl this component is *not* going to be used).
+dnl
+dnl $1: action if this component can compile
+dnl (we still invoke $1 so that "make distclean" and friends will work)
+dnl
+AC_DEFUN([MCA_opal_event_libevent2022_FAKE_CONFIG],[
+    MCA_opal_event_libevent2022_SUB_CONFIGURE([], [], [])
+    AC_MSG_NOTICE([remember: event:external will be used; this component was configured, but will be skipped])
+    $1
+])
 
-    AC_CONFIG_FILES([opal/mca/event/libevent2022/Makefile])
-    libevent_basedir="opal/mca/event/libevent2022"
+dnl
+dnl This macro has a bunch of side effects.  It is only meant to be
+dnl invoked when this component is going to be used (i.e., when
+dnl event:external is *not* going to be used).  If this macro is invoked
+dnl when event:external is used, Terrible Things will happen.
+dnl
+dnl $1: action if this component can compile
+dnl $2: action if this component cannot compile
+dnl
+AC_DEFUN([MCA_opal_event_libevent2022_REAL_CONFIG],[
+    OPAL_VAR_SCOPE_PUSH([CFLAGS_save CPPFLAGS_save libevent_file event_args libevent_happy])
 
     CFLAGS_save="$CFLAGS"
     CFLAGS="$OPAL_CFLAGS_BEFORE_PICKY $OPAL_VISIBILITY_CFLAGS"
@@ -175,14 +197,14 @@ AC_DEFUN([MCA_opal_event_libevent2022_BACKEND_CONFIG],[
 
     AC_MSG_RESULT([$event_args])
 
+    # Invoke the embedded configure script.
     # We define "random" to be "opal_random" so that Libevent will not
     # use random(3) internally (and potentially unexpectedly perturb
     # values returned by rand(3) to the application).
-
     CPPFLAGS="$CPPFLAGS -Drandom=opal_random"
-    OPAL_CONFIG_SUBDIR([$libevent_basedir/libevent],
-        [$event_args $opal_subdir_args 'CPPFLAGS=$CPPFLAGS'],
-        [libevent_happy="yes"], [libevent_happy="no"])
+    MCA_opal_event_libevent2022_SUB_CONFIGURE([$event_args],
+                                              [libevent_happy="yes"],
+                                              [libevent_happy="no"])
     if test "$libevent_happy" = "no"; then
         AC_MSG_WARN([Event library failed to configure])
         AC_MSG_ERROR([Cannot continue])
@@ -200,26 +222,34 @@ AC_DEFUN([MCA_opal_event_libevent2022_BACKEND_CONFIG],[
     # the value in the generated libevent/config.h (NOT
     # libevent/include/event2/event-config.h!).  Otherwise, set it to
     # 0.
-    libevent_file=$libevent_basedir/libevent/config.h
+    libevent_file=$opal_event_libevent2022_basedir/libevent/config.h
 
-    # If we are not building the internal libevent, then indicate that
-    # this component should not be built.  NOTE: we still did all the
-    # above configury so that all the proper GNU Autotools
-    # infrastructure is setup properly (e.g., w.r.t. SUBDIRS=libevent in
-    # this directory's Makefile.am, we still need the Autotools "make
-    # distclean" infrastructure to work properly).
-
-    AS_IF([test "$with_libevent" != "internal" && test -n "$with_libevent" && test "$with_libevent" != "yes"],
-          [AC_MSG_WARN([using an external libevent; disqualifying this component])
-           libevent_happy=no],
-
-          [AS_IF([test "$libevent_happy" = "yes" && test -r $libevent_file],
-            [OPAL_HAVE_WORKING_EVENTOPS=`grep HAVE_WORKING_EVENTOPS $libevent_file | awk '{print [$]3 }'`
-              $1],
-            [$2
-              OPAL_HAVE_WORKING_EVENTOPS=0])
-          ]
-    )
+    AS_IF([test "$libevent_happy" = "yes" && test -r $libevent_file],
+          [OPAL_HAVE_WORKING_EVENTOPS=`grep HAVE_WORKING_EVENTOPS $libevent_file | awk '{print [$]3 }'`
+           $1],
+          [$2
+           OPAL_HAVE_WORKING_EVENTOPS=0])
 
     OPAL_VAR_SCOPE_POP
+])
+
+dnl Call configure in the embedded libevent.
+dnl
+dnl We still do this so that all the proper GNU Autotools
+dnl infrastructure is setup properly (e.g., w.r.t. SUBDIRS=libevent in
+dnl this directorys Makefile.am, we still need the Autotools "make
+dnl distclean" infrastructure to work properly).
+dnl
+dnl $1: extra configure arguments
+dnl $2: action on success
+dnl $3: action on failure
+dnl
+AC_DEFUN([MCA_opal_event_libevent2022_SUB_CONFIGURE],[
+    # We define "random" to be "opal_random" so that Libevent will not
+    # use random(3) internally (and potentially unexpectedly perturb
+    # values returned by rand(3) to the application).
+
+    OPAL_CONFIG_SUBDIR([$opal_event_libevent2022_basedir/libevent],
+        [$1 $opal_subdir_args 'CPPFLAGS=$CPPFLAGS'],
+        [$2], [$3])
 ])

--- a/opal/mca/event/libevent2022/configure.m4
+++ b/opal/mca/event/libevent2022/configure.m4
@@ -24,7 +24,8 @@ AC_DEFUN([MCA_opal_event_libevent2022_COMPILE_MODE], [
 ])
 
 AC_DEFUN([MCA_opal_event_libevent2022_POST_CONFIG], [
-    AM_CONDITIONAL(OPAL_EVENT_HAVE_THREAD_SUPPORT, test "$enable_event_thread_support" = "yes")
+    AM_CONDITIONAL(OPAL_EVENT_HAVE_THREAD_SUPPORT,
+                   [test "$enable_event_thread_support" = "yes"])
     AS_IF([test "$1" = "1"],
           [ # Build libevent/include/event2/event-config.h.  If we
            # don't do it here, then libevent's Makefile.am will build

--- a/opal/mca/hwloc/hwloc201/configure.m4
+++ b/opal/mca/hwloc/hwloc201/configure.m4
@@ -69,27 +69,23 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_POST_CONFIG],[
 # MCA_hwloc_hwloc201_CONFIG([action-if-found], [action-if-not-found])
 # --------------------------------------------------------------------
 AC_DEFUN([MCA_opal_hwloc_hwloc201_CONFIG],[
-    # We know that the external hwloc component will be configured
-    # before this one because of its priority.  This component is only
-    # needed if the external component was not successful in selecting
-    # itself.
-    AC_MSG_CHECKING([if hwloc external component succeeded])
-    AS_IF([test "$opal_hwloc_external_support" = "yes"],
-          [AC_MSG_RESULT([yes])
-           AC_MSG_NOTICE([hwloc:external succeeded, so this component will be skipped])
-           $2],
-          [AC_MSG_RESULT([no])
-           AC_MSG_NOTICE([hwloc:external failed, so this component will be used])
-           MCA_opal_hwloc_hwloc201_BACKEND_CONFIG($1, $2)])
-])
-
-AC_DEFUN([MCA_opal_hwloc_hwloc201_BACKEND_CONFIG],[
     # Hwloc needs to know if we have Verbs support
     AC_REQUIRE([OPAL_CHECK_VERBS_DIR])
 
     AC_CONFIG_FILES([opal/mca/hwloc/hwloc201/Makefile])
 
     OPAL_VAR_SCOPE_PUSH([HWLOC_VERSION opal_hwloc_hwloc201_save_CPPFLAGS opal_hwloc_hwloc201_save_LDFLAGS opal_hwloc_hwloc201_save_LIBS opal_hwloc_hwloc201_save_cairo opal_hwloc_hwloc201_save_xml opal_hwloc_hwloc201_save_mode opal_hwloc_hwloc201_basedir opal_hwloc_hwloc201_file opal_hwloc_hwloc201_save_cflags CPPFLAGS_save LIBS_save opal_hwloc_external])
+
+    # We know that the external hwloc component will be configured
+    # before this one because of its priority.  This component is only
+    # needed if the external component was not successful in selecting
+    # itself.  Print out a message explaining this.
+    AC_MSG_CHECKING([if hwloc external component succeeded])
+    AS_IF([test "$opal_hwloc_external_support" = "yes"],
+          [AC_MSG_RESULT([yes])
+           AC_MSG_NOTICE([hwloc:external succeeded, so this component will be configured, but then will be skipped])],
+          [AC_MSG_RESULT([no])
+           AC_MSG_NOTICE([hwloc:external failed, so this component will be used])])
 
     # default to this component not providing support
     opal_hwloc_hwloc201_basedir=opal/mca/hwloc/hwloc201
@@ -98,8 +94,6 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_BACKEND_CONFIG],[
     opal_hwloc_hwloc201_save_CPPFLAGS=$CPPFLAGS
     opal_hwloc_hwloc201_save_LDFLAGS=$LDFLAGS
     opal_hwloc_hwloc201_save_LIBS=$LIBS
-
-    HWLOC_SET_SYMBOL_PREFIX([opal_hwloc201_])
 
     # save XML or graphical options
     opal_hwloc_hwloc201_save_cairo=$enable_cairo
@@ -143,6 +137,12 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_BACKEND_CONFIG],[
     AS_IF([test -n "$opal_datatype_cuda_CPPFLAGS"],
           [CPPFLAGS="$CPPFLAGS $opal_datatype_cuda_CPPFLAGS"])
 
+    # Only set the symbol prefix if this component is being used
+    # (i.e., if the external component is not being used).
+    AS_IF([test "$opal_hwloc_external_support" = "no"],
+          [HWLOC_SET_SYMBOL_PREFIX([opal_hwloc201_])])
+
+    # Do the bulk of the hwloc core setup
     HWLOC_SETUP_CORE([opal/mca/hwloc/hwloc201/hwloc],
               [AC_MSG_CHECKING([whether hwloc configure succeeded])
                AC_MSG_RESULT([yes])
@@ -202,8 +202,8 @@ AC_DEFUN([MCA_opal_hwloc_hwloc201_BACKEND_CONFIG],[
     # infrastructure is setup properly (e.g., w.r.t. SUBDIRS=hwloc in
     # this directory's Makefile.am, we still need the Autotools "make
     # distclean" infrastructure to work properly).
-    AS_IF([test "$opal_hwloc_external" = "yes"],
-          [AC_MSG_WARN([using an external hwloc; disqualifying this component])
+    AS_IF([test "$opal_hwloc_external_support" = "yes"],
+          [AC_MSG_NOTICE([using an external hwloc; disqualifying this component])
            opal_hwloc_hwloc201_support=no],
           [AC_DEFINE([HAVE_DECL_HWLOC_OBJ_OSDEV_COPROC], [1])
            AC_DEFINE([HAVE_HWLOC_TOPOLOGY_DUP], [1])])

